### PR TITLE
GNN tests

### PR DESCRIPTION
**Pull request by nicelhc13**
This PR fixes GNN CPU and single class tests. These fixes are done by hand calculation mostly.

### CODE REVIEW

**Comment by patrickkenney9801** [comment]
Please add all the inputs to git lfs instead of normal git and rebase on master

**Comment by patrickkenney9801** [comment]
Also more importantly you need to add at least some of these tests to this make directive so that they are run on ci here
https://github.com/utcs-scea/Galois/blob/c54db229836e759b8c87792c6933bf780d2f021d/Makefile#L116

**Comment by nicelhc13** [comment]
Done. Please review this.

File: inputs/CMakeLists.txt [hunk 1]
```diff
@@ -1,17 +1,3 @@
...
-OUTPUT small_inputs
-COMMAND ${CMAKE_COMMAND} -E tar xJf lonestar-cpu-inputs.tar.gz
-DEPENDS lonestar-cpu-inputs.tar.gz
-WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/inputs
-COMMENT "Unpacking lonestar-cpu-inputs.tar.gz"
-VERBATIM
-)
-
-add_custom_target(input DEPENDS small_inputs)
+version https://git-lfs.github.com/spec/v1
```
Comment (patrickkenney9801) [inline inputs/CMakeLists.txt]: why does this CMakeLists file look like this?
I would recommend just removing this file or reverting the changes may be better

**Comment by patrickkenney9801** [comment]
the files have not been added to git lfs either it seems, you probably have to remove them from git and then add them back to lfs
you also need to run `make pre-commit` on the repo

File: inputs/CMakeLists.txt (hunk 1 above)
Comment (nicelhc13) [reply inputs/CMakeLists.txt]: Oops. Thank you for catching this. It seemed like git-lfs changed it.

File: inputs/.gitattributes [hunk 2]
```diff
@@ -0,0 +1,5 @@
+cholesky filter=lfs diff=lfs merge=lfs -text
+CMakeLists.txt filter=lfs diff=lfs merge=lfs -text
```
Comment (patrickkenney9801) [inline inputs/.gitattributes]: this line should be removed

Comment (patrickkenney9801) [reply inputs/.gitattributes]: maybe via `git lfs rm` or similar? I am not sure

Comment (nicelhc13) [reply inputs/.gitattributes]: i mistakenly generated lfs attributes within inputs. I just recreated lfs from scratch.. something weird things went on. hope now it is resolved

**Comment by patrickkenney9801** [comment]
I think your `clang-format` may be too old, please check it is between versions 14 and 17, see https://github.com/utcs-scea/Galois/blob/c54db229836e759b8c87792c6933bf780d2f021d/Dockerfile#L80

**Comment by patrickkenney9801** [comment]
The lfs stuff all looks like it is working

**Comment by nicelhc13** [comment]
oh was this correct on your end? on my end, git lfs ls-files shows empty. looking into it... hmm

**Comment by patrickkenney9801** [comment]
> oh was this correct on your end? on my end, git lfs ls-files shows empty. looking into it... hmm

Sorry my bad, you are right the files are not added to `git lfs` -- but they are no longer present in git so that is something

**Comment by patrickkenney9801** [comment]
Try adding all `.gr`, `.csv`, `.bin`, etc files to `git lfs` I think that works better

**Comment by nicelhc13** [comment]
ok. i found why. i made a mistake. when I tracked those files, I just specified the directories since I expected that it recursively pushed all files in that directory. But it was not and I needed to specify it through **.

**Comment by patrickkenney9801** [comment]
> ok. i found why. i made a mistake. when I tracked those files, I just specified the directories since I expected that it recursively pushed all files in that directory. But it was not and I needed to specify it through **.

Interesting, thanks for letting me know that will come in handy in the future

**Comment by nicelhc13** [comment]
BTW I don't use docker and just manually installed clang-format-15 and applied that. I am still seeing a bunch of formatting updates. Did you also apply pre-commit and push the updates to master? (Because if you did, my updates look suspicious..)

**Comment by nicelhc13** [comment]
Other than that, I reflected your comments! Thanks!

**Comment by nicelhc13** [comment]
Oops. btw it seems like the server running ci/cd also needs to install intel packages. I am also seeing precommit error with clang-format 15.

**Comment by patrickkenney9801** [comment]
> Oops. btw it seems like the server running ci/cd also needs to install intel packages. I am also seeing precommit error with clang-format 15.

I checked and personally I have `clang-format 16`
I believe that ci/cd has the needed intel packages: https://github.com/utcs-scea/Galois/blob/c54db229836e759b8c87792c6933bf780d2f021d/Dockerfile#L37
Or is that insufficient?  If so please send a list of needed packages.

Given that that should be sufficient, it is probably an issue with https://github.com/utcs-scea/Galois/blob/c54db229836e759b8c87792c6933bf780d2f021d/Dockerfile#L86 not being run on container startup
Try changing https://github.com/utcs-scea/Galois/blob/c54db229836e759b8c87792c6933bf780d2f021d/.github/workflows/build-and-test.yml#L110
to be `CONTAINER_CMD="bash -lc 'source /opt/intel/oneapi/setvars.sh && make run-tests'" \`

**Comment by nicelhc13** [comment]
done

File: libgnn/test/accuracy-test.cpp [hunk 3]
```diff
@@ -5,6 +5,8 @@
 #include "galois/Logging.h"
 #include "galois/GraphNeuralNetwork.h"
 
+#define TESTER_DIR TESTINPUT "/tester/"
```
Comment (patrickkenney9801) [inline libgnn/test/accuracy-test.cpp]: if you override this as `${SRC_DIR}/inputs` then the next subpath `tester` is lost so the paths are messed up

I would recommend using definitions for both cora and tester or change the tests to add that subpath themselves

Comment (nicelhc13) [reply libgnn/test/accuracy-test.cpp]: Thank you!

**Comment by nicelhc13** [comment]
Regarding clang-format, I've noticed that I've used 18. I am trying to downgrade its version.

**Comment by nicelhc13** [comment]
ok.. finally passed precommit. i tested 14, 15, 16, 17, 18. Only 16.0.6. worked. I think we need to add rules and generalize the versions?

**Comment by nicelhc13** [comment]
Plus, is it possible to add timeout? It seems like the current timeout is not a bug but just slow?

**Comment by patrickkenney9801** [comment]
> ok.. finally passed precommit. i tested 14, 15, 16, 17, 18. Only 16.0.6. worked. I think we need to add rules and generalize the versions?

Thank you for going through all of that -- we definitely should generalize it so it is not such a pain point in the future

Unfortunately I do not know enough about the tool to do that right now but I will look into it.

For now we can just tell people to use `16.0.6` if they have issues.

**Comment by patrickkenney9801** [comment]
> Details

Unfortunately the line `Test timeout computed to be: 10000000` does not mean the tests timed out, I am unsure what the issue is there

**Comment by patrickkenney9801** [comment]
> > Details
> 
> Unfortunately the line `Test timeout computed to be: 10000000` does not mean the tests timed out, I am unsure what the issue is there

It looks like the issue manifests when you run with more than 6 processes, if these tests pass for you locally I would recommend disabling them for ci

**Comment by nicelhc13** [comment]
This CI/CD tests kept failing due to the input graph path. The path that the release version worked doesn't work for the debug version... Do you have any idea? Is it possible to access the server running this CI/CD and check the path on my end? Thank you.

**Comment by patrickkenney9801** [comment]
> This CI/CD tests kept failing due to the input graph path. The path that the release version worked doesn't work for the debug version... Do you have any idea? Is it possible to access the server running this CI/CD and check the path on my end? Thank you.

CI/CD is running on zerberus - but everything you care about would be running in containers

**Comment by patrickkenney9801** [comment]
> Details

It looks like the debug paths are correct

My guess would be that there is some `#ifndef NDEBUG` or `assert` or `GALOIS_ASSERT` or other debug logic that is throwing the errors that is not run when built via Release

**Comment by nicelhc13** [comment]
Thank you for your comment. But I cannot reproduce this on my end on both Release and Debug mode. 
`terminate called after throwing an instance of 'char const*'`
This error in general came from an incorrect directory path.
My confusion is the log printing the path looks correct, but it keeps raising this error during file reading at CuSP side.

File: libgnn/test/CMakeLists.txt [hunk 4]
```diff
@@ -1,7 +1,11 @@
 find_package(OpenMP)
 
+set(TESTINPUT "${CMAKE_SOURCE_DIR}/inputs")
```
Comment (patrickkenney9801) [inline libgnn/test/CMakeLists.txt]: please change this to `Galois_SOURCE_DIR`

File: .github/workflows/build-and-test.yml [hunk 5]
```diff
@@ -107,7 +107,7 @@ jobs:
     - name: Run Tests
       timeout-minutes: 5
```
Comment (patrickkenney9801) [inline .github/workflows/build-and-test.yml]: please bump this to 15 minutes

**Comment by patrickkenney9801** [comment]
try running this with the `Debug` build disabled, remove it from this line https://github.com/utcs-scea/Galois/blob/ede08b64db82478a698fc6e844297a70dd65da68/.github/workflows/build-and-test.yml#L56
and then also remove the `exclude` block below it

**Comment by patrickkenney9801** [comment]
also please change every `/pando-galois` to just `/galois` from https://github.com/utcs-scea/Galois/blob/ede08b64db82478a698fc6e844297a70dd65da68/.github/workflows/build-and-test.yml#L45

**Comment by nicelhc13** [comment]
Thanks. Just in case, I pushed file read checking code. I will follow your instruction too.

**Comment by nicelhc13** [comment]
input path check at GNNGraph.hpp has been passed. Something is happening within CuSP. I am thinking how to reproduce or check this failure.

File: .github/workflows/build-and-test.yml [hunk 6]
```diff
@@ -55,9 +55,9 @@ jobs:
       matrix:
         build-type: ['Release', 'Debug']
```
Comment (patrickkenney9801) [inline .github/workflows/build-and-test.yml]: you also need to remove `'Debug'` from here

**Comment by nicelhc13** [comment]
why doesnt this have ninja

File: Makefile [hunk 7]
```diff
@@ -4,13 +4,13 @@ UNAME ?= $(shell whoami)
 UID ?= $(shell id -u)
 GID ?= $(shell id -g)
 
-BASE_IMAGE_NAME ?= pando-galois
+BASE_IMAGE_NAME ?= galois
```
Comment (patrickkenney9801) [inline Makefile]: for now keep this one `pando-galois`

**Comment by patrickkenney9801** [comment]
the latest culprit for the last few errors is the docker image env vars, you'll have to change those too https://github.com/utcs-scea/Galois/blob/1c0e88e5db31a6e60a912a7e02a562dbd07b256d/Dockerfile#L65
or revert the last changes of `/pando-galois` -> `/galois`

**Review by patrickkenney9801 — APPROVED** [review]
Looks great, thanks for all the work fighting the system

**Pull request merged**